signac/contrib/project.py

```diff
@@ -372,7 +372,7 @@ def open_job(self, statepoint=None, id=None):
             # second best case
             job = self.Job(project=self, statepoint=statepoint)
             if job._id not in self._sp_cache:
-                self._sp_cache[job._id] = dict(job._statepoint)
+                self._sp_cache[job._id] = job._statepoint._as_dict()
```

Comment:
I might suggest using `job.statepoint` here for future proofing against any change to internals.

Reply:
I think I used `._statepoint` to avoid the extra function call caused by the `@property`, but hopefully that does not really make any difference.